shards/rust/src/types.rs

```diff
@@ -4012,18 +4038,18 @@ impl ParamVar {
     // notice we don't need to fix up ref-counting because cloning does not touch that
   }
 
-  pub fn get(&self) -> &Var {
+  pub fn get(&self) -> &'static Var {
     assert_ne!(self.pointee, std::ptr::null_mut());
     unsafe { &*self.pointee }
   }
 
   // Users should never fully overwrite or flags will be lost, unless taken care of
-  pub fn get_mut(&mut self) -> &mut Var {
+  pub fn get_mut(&mut self) -> &'static mut Var {
```

Comment:
Don't change ParamVar for this. use referenceVariable directly in get_or_var

Reply:
Additionally define a placeholder for findVariable that doesn't increase refcount which can later be in the core interface